lib/specifications/ci-types/jenkins/results.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Methods to retrieve the results from a given jenkins server when supplied with URLs to job locations
+ */
+'use strict';
+
+var XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest;
+var appConfig = require('../../../configuration/app-config').get();
+
+module.exports = {
+  getResults: getResults
+};
+
+
+/**
+ * Loops through the array of jenkins URLs to get the job names and then scenario names to be used to display the
+ * results on the feature pages.
+ */
+function getResults(projectData) {
+  var j;
+  for(j=0;j<projectData.config.jenkinsJobs.length;j++) {
+    httpGetJobs(projectData.config.jenkinsJobs[j]);
+  }
+}
+
+/**
+ * Hits a url, calls a call back function and returns an array
+ */
+function httpGet(theUrl, callBack, async) {
+  var xmlHttp = new XMLHttpRequest();
```

Comment:
I would recommend using https://www.npmjs.com/package/request-promise instead of XMLHttpRequest, it will make it easier to query external sources